feat: kdoc export by default (#5027)
JetBrains/kotlin-web-site: docs/topics/native/native-objc-interop.md

+ (void)printSumA:(int32_t)a b:(int32_t)b __attribute__((swift_name("printSum(a:b:)")));
```

KDoc comments are embedded into klibs and extracted from klibs into the produced Apple frameworks.
As a result, comments on classes and methods appear during autocompletion, for example in Xcode.

> If you go to the definition of functions (in the `.h` file), you'll see comments on `@param`, `@return`, and so on.

Reviewer: Do we need the parentheses here?
If you go to the definition of functions in the `.h` file, you'll

> If you go to the definition of functions (in the `.h` file), you'll see comments on `@param`, `@return`, and so on.

Reviewer: ```suggestion
If you go to the definition of functions (in the `.h` file), you'll see comments on `@param`, `@return`, and similar tags.
```

Reviewer: again just a suggestion - I think it works as it is, but perhaps a bit clearer? 🤔

Known limitations:

* Dependency documentation is not exported unless it is compiled with the `-Xexport-kdoc` option. Libraries compiled with
  this compiler option might be incompatible with other compiler versions.

> * KDoc comments are mostly exported as is. Many KDoc features, for example `@property`, are not supported.

Reviewer: ```suggestion
* KDoc comments are mostly exported as is, but many KDoc block tags, such as `@property`, are not supported.
```

Reviewer: (suggestion)
not part of the update but perhaps a bit more natural this way

> If necessary, you can disable the export of Kdoc comments from klibs to the produced Apple frameworks in the `binaries {}`

Reviewer: ```suggestion
If necessary, you can disable the export of KDoc comments from klibs to the produced Apple frameworks in the `binaries {}`
```
